src/Illuminate/Container/Container.php

```diff
@@ -261,7 +261,7 @@ protected function getClosure($abstract, $concrete)
                 return $container->build($concrete);
             }
 
-            return $container->resolve(
+            return $container->make(
```

Comment:
The make method is just an alias of resolve. This can't solve anything really.

Reply:
It solves the issue. Try checking out previous commit where test was added. Resolve is an alias of make in `Illuminate\Container\Container`, but not in `Illuminate\Foundation\Application`.

Reply:
@driesvints it actually could in the context of an application because the Foundation Application overrides `make` to do the deferred provider stuff.